pkg/sdk/testint/external_volumes_gen_integration_test.go

```diff
@@ -0,0 +1,605 @@
+		StorageProvider         string   `json:"STORAGE_PROVIDER"`
+		StorageBaseUrl          string   `json:"STORAGE_BASE_URL"`
+		StorageAllowedLocations []string `json:"STORAGE_ALLOWED_LOCATIONS"`
+		StorageAwsRoleArn       string   `json:"STORAGE_AWS_ROLE_ARN"`
+		StroageAwsIamUserArn    string   `json:"STORAGE_AWS_IAM_USER_ARN"`
+		StorageAwsExternalId    string   `json:"STORAGE_AWS_EXTERNAL_ID"`
+		EncryptionType          string   `json:"ENCRYPTION_TYPE"`
+		EncryptionKmsId         string   `json:"ENCRYPTION_KMS_KEY_ID"`
+	}
+
+	type S3StorageLocationTrimmed struct {
+		Name                 string `json:"NAME"`
+		StorageProvider      string `json:"STORAGE_PROVIDER"`
+		StorageBaseUrl       string `json:"STORAGE_BASE_URL"`
+		StorageAwsRoleArn    string `json:"STORAGE_AWS_ROLE_ARN"`
+		StorageAwsExternalId string `json:"STORAGE_AWS_EXTERNAL_ID"`
+		EncryptionType       string `json:"ENCRYPTION_TYPE"`
+		EncryptionKmsId      string `json:"ENCRYPTION_KMS_KEY_ID,omitempty"`
+	}
+
+	type GCSStorageLocation struct {
+		Name                     string   `json:"NAME"`
+		StorageProvider          string   `json:"STORAGE_PROVIDER"`
+		StorageBaseUrl           string   `json:"STORAGE_BASE_URL"`
+		StorageAllowedLocations  []string `json:"STORAGE_ALLOWED_LOCATIONS"`
+		StorageGcpServiceAccount string   `json:"STORAGE_GCP_SERVICE_ACCOUNT"`
+		EncryptionType           string   `json:"ENCRYPTION_TYPE"`
+		EncryptionKmsId          string   `json:"ENCRYPTION_KMS_KEY_ID"`
+	}
+
+	type GCSStorageLocationTrimmed struct {
+		Name            string `json:"NAME"`
+		StorageProvider string `json:"STORAGE_PROVIDER"`
+		StorageBaseUrl  string `json:"STORAGE_BASE_URL"`
+		EncryptionType  string `json:"ENCRYPTION_TYPE"`
+		EncryptionKmsId string `json:"ENCRYPTION_KMS_KEY_ID,omitempty"`
+	}
+
+	type AzureStorageLocation struct {
+		Name                    string   `json:"NAME"`
+		StorageProvider         string   `json:"STORAGE_PROVIDER"`
+		StorageBaseUrl          string   `json:"STORAGE_BASE_URL"`
+		StorageAllowedLocations []string `json:"STORAGE_ALLOWED_LOCATIONS"`
+		AzureTenantId           string   `json:"AZURE_TENANT_ID"`
+		AzureMultiTenantAppName string   `json:"AZURE_MULTI_TENANT_APP_NAME"`
+		AzureConsentUrl         string   `json:"AZURE_CONSENT_URL"`
+		EncryptionType          string   `json:"ENCRYPTION_TYPE"`
+		EncryptionKmsId         string   `json:"ENCRYPTION_KMS_KEY_ID"`
+	}
+
+	type AzureStorageLocationTrimmed struct {
+		Name            string `json:"NAME"`
+		StorageProvider string `json:"STORAGE_PROVIDER"`
+		StorageBaseUrl  string `json:"STORAGE_BASE_URL"`
+		AzureTenantId   string `json:"AZURE_TENANT_ID"`
+	}
+
+	// Enforce only property names and values in tests, not parent_property, type and property_default
+	// In addition the storage location properties are trimmed to only contain values that we set
+	trimProperties := func(t *testing.T, props []sdk.ExternalVolumeProperty) []ExternalVolumePropNameValue {
```

Comment:
Nice function! :)

I think, that ultimately it will have to be added to the production code of the SDK, so that we need an easier access to these values from the terraform resource level. If it happens, it will also require a few unit tests to ensure it handles all the corner case correctly. It will also need a few modifications (like removing the panics) first.

But I would leave it for now as is, but keeping it in mind. cc: @sfc-gh-jcieslak